app/code/core/Mage/Catalog/Model/Category.php

```diff
@@ -918,10 +918,16 @@ public function getAvailableSortByOptions() {
      * @return string
      */
     public function getDefaultSortBy() {
-        if (!$sortBy = $this->getData('default_sort_by')) {
+
+        $sortBy = $this->getData('default_sort_by');
+
+        //when not set or not available use default from system config
+        if (!$sortBy || !isset($available[$sortBy])) {
```

Comment:
Variable `$available` is not defined yet

Reply:
agree code needs logic update by OP

Reply:
Dunno if this violates PSR or not but I often use the assignment in the if statement but it has to be wrapped in parentheses:

```
if ( ! ($sortBy = $this->getData('default_sort_by'))) {
```

Reply:
fixed now, the $available variable is initialized on the right place